### PR TITLE
User profile

### PR DESCRIPTION
**Pull request by jasonify**

### CODE REVIEW

File: server/controllers/auth.controller.js [hunk 1]
```diff
@@ -110,8 +110,10 @@ function register(req, res, next) {
       }
 
       const newUser = new User();
-      newUser.username = username;
-      newUser.password = newUser.generateHash(password);
+      newUser.password = User.generateHash(password);
+      // We assing a set of "approved fields"
```
Comment (crablar) [inline server/controllers/auth.controller.js]: assign

File: config/param-validation.js [hunk 2]
```diff
@@ -43,7 +19,14 @@ export default {
   login: {
     body: {
       username: Joi.string().required(),
-      password: Joi.string().required()
+      password: Joi.string().required(),
+      // Should be required once mobile apps get updated:
+      // TODO: consider adding versioning to API so we can roll
+      // out new features to different clients.
+      name: Joi.string(),
+      bio: Joi.string(),
```
Comment (crablar) [inline config/param-validation.js]: Why are bio and website parameters for login?

Comment (jasonify) [reply config/param-validation.js]: Ah, nice catch.

File: server/controllers/auth.controller.js (hunk 1 above)
Comment (jasonify) [reply server/controllers/auth.controller.js]: fixed :)

**Pull request merged**